front/app/utils/cl-react-query/fetcher.ts

```diff
@@ -206,6 +206,9 @@ async function fetcher({
       }
     }
   }
+
+  if (!data) return null;
```

Comment:
Why is this needed?

Reply:
This is related to a change in behavior from `msw`. Before, in some of our mock server setups, we would do something like `return res(ctx.status(201));` (e.g. [here](https://github.com/CitizenLabDotCo/citizenlab/blob/master/front/app/api/analysis_background_tasks/useLaunchAnalysisAutotagging.test.ts#L13)). What we do here is return a status code indicating a successful request, but with a nullish response.

In our `fetcher`, this was seen as invalid somehow- it threw an error on `await response.json();`, which was then caught and handled. 

After running the codemod, all instances of this were replaced by `return HttpResponse.json(null, { status: 201 });`. But now, the `fetcher` started behaving differently- the `await response.json();` call doesn't throw an error, but actually executes fine. The only difference is that `data` is now `null`. This is not a problem until you start destructuring it at the bottom of the file.

I couldn't figure out how to reproduce the old behavior, but when you think about it, the new behavior seems much better- sometimes, receiving `null` back from the server is fine and should trigger an error. Also, if we want to tweak this behavior we have to go back and fix like a 100 files manually... so I figured it's fine like this

Reply:
I think there was a reason for this behaviour when we were setting this up but I no longer remember what is was or whether it's still relevant. So as long as you are confident that this really doesn't break anything, I'm ok do go with it. 👍 Let's make sure we take all possible precautions though - run the e2e tests, etc.